### DISCUSSION TITLE
Stipple.init and the life cycle of model instances?

### DISCUSSION THREAD

**Discussion by michaelfliegner** [GenieFramework/Stipple.jl#99] [Q&A] [answered]
I would want to display a model on different pages. But changing the page - by using a form for instance - resets the model fields to initial values. So ... if the model instance "dies" when the page changes, then I would want

- my model values to survive server side and
- a device to pass a survived model instance to the renderer (Stipple ].init can't do that, I guess)

PLS help!

**my code:**
```julia
model=handlers(Stipple.init(Model; transport = Genie.WebThreads))
ctx=@__MODULE__

route("/") do
    println(ctx)
    html(ui(model),context=ctx )
end

route("/sub", method = POST) do
    age = params(:age, "99")
    name = params(:name, "Schnipperdey")
    println("PARAMS " * name * "  " * string(age))
    redirect("/")
end
```

**Comment by essenciary** [comment, marked as answer]
@michaelfliegner It looks like you instantiate the model as a global, so it should survive on the server side between pages. 

The design pattern to generate the state on the client consists of 2 steps: 
* instantiating a default state on the client side (so that the JS file is unique, can be cached and hosted on CDNs, and does not leak user state) 
*  pushing whatever model state you have from the server when the frontend app is ready by listening to the `isready` event. 

This means that you need to add the following to your code: 

```julia
on(model.isready) do _
  push!(model)
end
```

**Reply by jochenkrattenmacher** [reply]
Hello,
I just found this, and I am thinking since this seems to be pretty fundamental about Stipple, it could go into the docs somewhere?

In the[ stipple API](https://www.genieframework.com/docs/stipple/api/stipple.html), there is an entry on "Base.push!", but well, it seems that is by accident, since this entry doesn't have much to do with Stipple?

> push!(collection, items...) -> collection
> 
> Insert one or more items in collection. If collection is an ordered container, the items are inserted at the end (in the given order).
> ...


I guess the interesting methods are the ones within Stipple.jl (starting with "function Base.push!(app::M,")?


I also wonder whether there are instances where one would prefer not to add the code as suggested here? In other words, why is it not standard behavior that the model state is pushed to the frontend app upon isready? This is probably a super newbie question and I'll find out soon enough, so I am not necessarily expecting an answer!


Cheers

**Reply by essenciary** [reply]
Unfortunately this is an issue with `Documenter.jl` -- when specializing an existing function, it will pick the documentation of the original function. I don't know how to fix it (or if it's at all fixable).

**Reply by AbhimanyuAryan** [reply]
I tried to look for this in existing Documenter.jl issues and google but no luck. I'll open an issue on Documenter.jl ask the team directly
